pkg/fleet-manager/migrate_controller.go

```diff
@@ -0,0 +1,292 @@
+	}
+
+	// Get the status of Velero backup resources
+	veleroBackup := &velerov1.Backup{}
+	err = getResourceFromClusterClient(ctx, sourceBackupName, VeleroNamespace, *sourceClusterAccess, veleroBackup)
+	if err != nil {
+		log.Error(err, "Failed to retrieve backup resource for migration", "backupName", sourceBackupName)
+		return ctrl.Result{}, fmt.Errorf("retrieving backup status: %w", err)
+	}
+
+	// Update migration status based on the Velero backup details
+	currentBackupDetails := &backupapi.BackupDetails{
+		ClusterName:           sourceClusterKey.Name,
+		ClusterKind:           sourceClusterKey.Kind,
+		BackupNameInCluster:   veleroBackup.Name,
+		BackupStatusInCluster: &veleroBackup.Status,
+	}
+	migrate.Status.SourceClusterStatus = currentBackupDetails
+
+	if veleroBackup.Status.Phase == velerov1.BackupPhaseCompleted {
+		conditions.MarkTrue(migrate, backupapi.SourceReadyCondition)
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// reconcileMigrateRestore handles the restore stage of the migration process.
+func (m *MigrateManager) reconcileMigrateRestore(ctx context.Context, migrate *backupapi.Migrate) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// If source cluster's backup resource is not ready, return directly.
+	if !isMigrateSourceReady(migrate) {
+		return ctrl.Result{RequeueAfter: RequeueAfter}, nil
+	}
+
+	targetClusters, err := fetchDestinationClusters(ctx, m.Client, migrate.Namespace, migrate.Spec.TargetClusters)
+	if err != nil {
+		log.Error(err, "Failed to fetch target clusters for migration")
+		return ctrl.Result{}, fmt.Errorf("fetching target clusters: %w", err)
+	}
+
+	if migrate.Status.Phase != backupapi.MigratePhaseInProgress {
+		migrate.Status.Phase = backupapi.MigratePhaseInProgress
+		log.Info("Migrate Phase changes", "phase", backupapi.MigratePhaseInProgress)
+		// Ensure the restore point is created after the backup.
+		time.Sleep(MigrationDelay)
+	}
+
+	// referredBackupName is same in different target clusters velero restore, because the velero backup will sync to current cluster.
+	// SourceCluster only has one cluster, so the cluster[0].name is the real name of SourceCluster
+	referredBackupName := generateVeleroResourceName(migrate.Spec.SourceCluster.Clusters[0].Name, MigrateKind, migrate.Namespace, migrate.Name)
+	restoreLabel := generateVeleroInstanceLabel(MigrateNameLabel, migrate.Name, migrate.Spec.TargetClusters.Fleet)
+	// Handle create target clusters' velero restore
+	var tasks []func() error
+	for clusterKey, clusterAccess := range targetClusters {
+		// Ensure the velero backup has been sync to current cluster
+		referredVeleroBackup := &velerov1.Backup{}
+		if err = getResourceFromClusterClient(ctx, referredBackupName, VeleroNamespace, *clusterAccess, referredVeleroBackup); err != nil {
+			if apierrors.IsNotFound(err) {
+				// if not found, requeue with `RequeueAfter`
+				return ctrl.Result{RequeueAfter: RequeueAfter}, nil
```

Comment:
In theory this could not happen

Reply:
the sync of velero backup may take time, so it could happen.  the velero backup need syn from source cluster to target cluster

Reply:
the velero backup is create in `source clusters`, so we need ensure that this can be found in each `target clusters`